desktop_version/src/Music.cpp

```diff
@@ -92,81 +131,263 @@ class SoundTrack
+            musicVoice = NULL;
+        }
     }
 
     static bool IsHalted()
     {
-        /* return musicVoice == NULL; */
-        return Mix_PausedMusic() == 1;
+        return musicVoice == NULL;
     }
 
     static void Pause()
     {
-        /* FAudioSourceVoice_Pause */
-        Mix_PauseMusic();
+        vlog_info("paused musictracks");
+        if (!IsHalted()) FAudioSourceVoice_Stop(musicVoice, 0, FAUDIO_COMMIT_NOW);
     }
 
     static void Resume()
     {
-        /* FAudioSourceVoice_Resume */
-        Mix_ResumeMusic();
+        vlog_info("resumed musictracks");
+        if (!IsHalted()) FAudioSourceVoice_Start(musicVoice, 0, FAUDIO_COMMIT_NOW);
     }
 
     static void SetVolume(int musicVolume)
     {
-        /* FAudioSourceVoice_SetVolume */
-        Mix_VolumeMusic(musicVolume);
+        float adj_vol = ((float)musicVolume/VVV_MAX_VOLUME)*2;
+        if (!IsHalted()) FAudioVoice_SetVolume(musicVoice, adj_vol, FAUDIO_COMMIT_NOW);
     }
 
 private:
-    Mix_Music *m_music;
-};
+    stb_vorbis* vorbis = NULL;
+    int channels;
+    Uint32 size;
+    int loopbegin;
+    int looplength;
+    int sample_pos; //stb_vorbis offset not yet functional on pulldata API. TODO Replace when fixed
+
+    FAudioVoiceCallback callbacks;
+    FAudioWaveFormatEx format;
+
+    Uint8* decoded_buf_playing = NULL;
+    Uint8* decoded_buf_reserve = NULL;
+    Uint8* read_buf = NULL;
+    bool shouldloop;
+
+    static FAudioSourceVoice* musicVoice;
+
+    static void refillReserve(FAudioVoiceCallback* callback, void* ctx) {
+        MusicTrack* t = (MusicTrack*)ctx;
+        FAudioBuffer faudio_buffer;
+        SDL_memset(&faudio_buffer, 0, sizeof(FAudioBuffer));
+        if (t->looplength == 0) {
+            faudio_buffer.PlayLength = stb_vorbis_get_samples_float_interleaved(t->vorbis, t->channels, (float*)t->decoded_buf_reserve, t->size/sizeof(float));
+        } else {
+            faudio_buffer.PlayLength = std::min(stb_vorbis_get_samples_float_interleaved(t->vorbis, t->channels, (float*)t->decoded_buf_reserve, t->size/sizeof(float)), (t->loopbegin+t->looplength)-t->sample_pos);
+        }
+        faudio_buffer.AudioBytes = t->size;
+        faudio_buffer.pAudioData = t->decoded_buf_reserve;
+        faudio_buffer.pContext = t;
+        if (faudio_buffer.PlayLength == 0) {
+            if (t->shouldloop) {
+                stb_vorbis_seek(t->vorbis, t->loopbegin);
+                t->sample_pos = t->loopbegin;
+                if (t->looplength != 0) {
+                    faudio_buffer.PlayLength = std::min(stb_vorbis_get_samples_float_interleaved(t->vorbis, t->channels, (float*)t->decoded_buf_reserve, t->size/sizeof(float)), t->looplength);
+                } else {
+                    faudio_buffer.PlayLength = stb_vorbis_get_samples_float_interleaved(t->vorbis, t->channels, (float*)t->decoded_buf_reserve, t->size/sizeof(float));
+                }
+                if (faudio_buffer.PlayLength == 0) return;
```

Comment:
I know that this is done elsewhere in the codebase, but I'd really like for this to be a proper if-block instead of a one-liner - e.g. please do this instead:
```c++
if (cond)
{
    statement;
}
```